python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -109,8 +109,11 @@ def _with_code_source_single_definition(
             else assets_def.node_def.compute_fn
         )
     elif isinstance(assets_def.node_def, GraphDefinition):
-        # todo - properly handle graph-backed asset code source
-        return assets_def
+        # For graph-backed assets, point to the composition fn, e.g. the
+        # function decorated by @graph_asset
+        base_fn = assets_def.node_def.composition_fn
+        if not base_fn:
+            return assets_def
```

Comment:
Just so things flow better

- We should instantiate `base_fn = None` at the start of this helper function.
- We should extract this early return outside of the conditional. Then we can get rid of the `else` statement altogether.